conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, CMake, tools
+
+
+class HelloConan(ConanFile):
+    name = "LibConfig"
+    version = "1.7.2"
+    license = "GNU LESSER GENERAL PUBLIC LICENSE"
+    author = """Mark Lindner - Lead developer & maintainer.
+                Daniel Marjamäki - Enhancements & bugfixes.
+                Andrew Tytula - Windows port.
+                Glenn Herteg - Enhancements, bugfixes, documentation corrections.s
+                Matt Renaud - Enhancements & bugfixes.
+                JoseLuis Tallon - Enhancements & bugfixes"""
+    url = "hyperrealm.github.io/libconfig/"
+    description = "Libconfig is a simple library for processing structured configuration files, like this one. This file format is more compact and more readable than XML. And unlike XML, it is type-aware, so it is not necessary to do string parsing in application code."
+    topics = ("config")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+    generators = "cmake"
+    exports_sources="*"
```

Comment:
```suggestion
    no_copy_source = True
    exports_sources="*"
```

Reply:
This line results in:

```
LibConfig/1.7.2: ERROR: Package 'd3058fa461095e3b567f10d3a72b4fac1de53cda' build failed
LibConfig/1.7.2: WARN: Build folder /home/lukas/.conan/data/LibConfig/1.7.2/_/_/build/d3058fa461095e3b567f10d3a72b4fac1de53cda
ERROR: LibConfig/1.7.2: Error in build() method, line 34
        self.cmake.configure(source_folder=self.build_folder)
        ConanException: Error 1 while executing cd '/home/lukas/.conan/data/LibConfig/1.7.2/_/_/build/d3058fa461095e3b567f10d3a72b4fac1de53cda' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="10" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/home/lukas/.conan/data/LibConfig/1.7.2/_/_/package/d3058fa461095e3b567f10d3a72b4fac1de53cda" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="OFF" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev '/home/lukas/.conan/data/LibConfig/1.7.2/_/_/build/d3058fa461095e3b567f10d3a72b4fac1de53cda'

what do you intend to do with this?

```

Reply:
Can you post the complete conan output?
I did give my suggestions without actually testing them on my system, so there will be bugs.

Reply:
I have checked out you PR and will open a pr on your fork.

Reply:
See https://github.com/Lukas-Heiligenbrunner/libconfig/pull/1

Reply:
Hey, 

if I do a `conan create .` on your PR code it results in an infinte loop (can you reproduce this?).
With the `no_copy_source` flag it builds correctly. This flag should only tell to not copy the sources, or am I wrong?

~Lukas

Reply:
The `no_copy_source` is essential.
Otherwise the wrapper cmake will overwrite libconfig's cmake script.
Which results in an infinite loop.

Reply:
The `no_copy_source` attribute is an instruction to conan to don't copy the files from the source directory to the build directory.
I think this is ok?

Reply:
Ah ok, this explains everything...  :D